src/Esfa.Vacancy.Application/Commands/CreateApprenticeship/CreateApprenticeshipParametersMapper.cs

```diff
@@ -55,14 +56,40 @@ public CreateApprenticeshipParametersMapper(IWageTypeMapper wageTypeMapper, IDur
                 EmployersApplicationInstructions = request.ExternalApplicationInstructions,
                 ContactName = request.ContactName,
                 ContactEmail = request.ContactEmail,
-                ContactNumber = request.ContactNumber
+                ContactNumber = request.ContactNumber,
+                TrainingType = (int)request.TrainingType,
+                ApprenticeshipType = GetApprenticeshipType(request.TrainingType, request.EducationLevel)
             };
 
             MapLocationFields(request, employerInformation, parameters);
 
+            MapTrainingCode(request, parameters);
             return parameters;
         }
 
+        private static ApprenticeshipType GetApprenticeshipType(TrainingType trainingType, int educationLevel)
+        {
+            switch (educationLevel)
+            {
+                case 2:
+                    return ApprenticeshipType.Intermediate;
+                case 3:
+                    return ApprenticeshipType.Advance;
+                case 4:
+                    return ApprenticeshipType.Higher;
+                case 5:
+                    return
+                        trainingType == TrainingType.Framework ? ApprenticeshipType.Degree : ApprenticeshipType.Foundation;
+                case 6:
+                    return ApprenticeshipType.Degree;
+                case 7:
+                    return
+                        trainingType == TrainingType.Framework ? ApprenticeshipType.Degree : ApprenticeshipType.Masters;
+                default:
+                    return (ApprenticeshipType)0;
```

Comment:
Do you need to cast this? `ApprenticeshipType.Unspecified` or whatever it is

Reply:
I am casting levels coming from Apprenticeship API to our internal ids, there is no case for unspecified.